node-repository/src/main/java/com/yahoo/vespa/hosted/provision/NodeRepository.java

```diff
@@ -568,7 +564,39 @@ public Node setDirty(Node node, Agent agent, String reason) {
             illegal("Could not deallocate " + nodeToDirty + ": " +
                     hostnamesNotAllowedToDirty + " are not in states [provisioned, failed, parked, breakfixed]");
 
-        return nodesToDirty.stream().map(node -> setDirty(node, agent, reason)).collect(Collectors.toList());
+        return nodesToDirty.stream().map(node -> deallocate(node, agent, reason)).collect(Collectors.toList());
+    }
+
+    /**
+     * Set a node dirty  or parked, allowed if it is in the provisioned, inactive, failed or parked state.
+     * Use this to clean newly provisioned nodes or to recycle failed nodes which have been repaired or put on hold.
+     *
+     * @throws IllegalArgumentException if the node has hardware failure
```

Comment:
Where is this thrown?

Reply:
I just moved this ... but you are right it doesn't look like it is thrown anywhere so I removed it.